docs/LibraryEvolution.rst

```diff
@@ -827,6 +839,7 @@ There are very few safe changes to make to protocols and their members:
 
 - A new non-type requirement may be added to a protocol, as long as it has an
   unconstrained default implementation.
+- A new associated type requirement may be added as long as it has a default.
```

Comment:
I'm not sure about this until we have a generalized existential story.

Reply:
Good point. This is true for adding new requirements with non-covariant Self, too. I'll add a separate bullet explaining that making something not usable as an existential is binary-compatible but not source-compatible.

Reply:
Yeah. Even if you already had associated types, I think we should be able to support the "existential of fully-constrained subprotocol" possibly before we get the full generalized existentials. It's a much easier change to reason about, and this would break that too.

Reply:
I have a radar on me for that :) One day...

Reply:
Amusing thing about binary compatibility: `as? P` casts written before `P` gained an associated type will continue to work fine, but the source code, if recompiled, would result in a compiler error.